roles/satellite-clone/files/disassociate_capsules.rb

```diff
@@ -11,42 +12,39 @@ def prepare_hammer_cmd(command)
 end
 
 def run_hammer_cmd(command)
-  command = prepare_hammer_cmd(command)
-  `#{command}`
+  `#{prepare_hammer_cmd(command)}`
 end
 
-def get_info_from_hammer(command, column=1)
-  bash_parse = " | grep -v \"Warning:\" | tail -n+2 | awk -F, {'print $#{column}'}"
-  run_hammer_cmd(command + bash_parse)
+def get_ids_from_hammer(command, search = nil)
+  JSON.parse(run_hammer_cmd("--output=json " + command + (search ? " --search #{search}" : "")).map do |obj|
```

Comment:
You should quote search since it contains spaces:
```suggestion
  JSON.parse(run_hammer_cmd("--output=json " + command + (search ? " --search '#{search}'" : "")).map do |obj|
```

Reply:
It's quoted when it's assigned a value, when the function is called at https://github.com/RedHatSatellite/satellite-clone/pull/392/files#diff-e156d7efb3320dd71eaac5bb78d7f6f2edd77b5aa05b767fbfdad959859fb463R28

And that seems to work fine:

```ruby
irb(main):043:0> search = 'feature = "Pulp Node"'
=> "feature = \"Pulp Node\""
irb(main):044:0> "--output=json " + "hammer capsule list" + (search ? " --search #{search}" : "")
=> "--output=json hammer capsule list --search feature = \"Pulp Node\""
```

(It also worked when I live tested these functions on Satellite server)

Reply:
Just kidding, I actually had this in my test environment but not here. I will correct the issue shortly.

Good catch. Thank you!

Reply:
Updated, it should be correct now:

```ruby
irb(main):045:0> search = 'feature = "Pulp Node"'
=> "feature = \"Pulp Node\""
irb(main):046:0> "--output=json " + "hammer capsule list" + (search ? " --search '#{search}'" : "")
=> "--output=json hammer capsule list --search 'feature = \"Pulp Node\"'"
```